src/utils.js

```diff
@@ -55,6 +55,14 @@ const mime_type = async (opts) => {
 const fetch_upload_data = async (opts) => {
   const { path, buffer } = opts;
 
+  if (path) {
+    try {
+      await fs.promises.access(path, fs.constants.F_OK | fs.constants.R_OK);
+    } catch (err) {
+      throw new Error(`Path ${path} does not exist or is not a readable file.`);
+    }
+  }
+
   const mime = await mime_type(opts);
   const data = path ? fs.createReadStream(path) : buffer;
   const size = path ? (await fs.promises.stat(path)).size : buffer.length;
```

Comment:
Indeed, running `stat` before `mime_type` wold grant us a similarly meaningful error with less code, but we wouldn't able to explicitly check if the path is a file nor if its contents are readable.

```suggestion
  const size = path ? (await fs.promises.stat(path)).size : buffer.length;
  const mime = await mime_type(opts);
  const data = path ? fs.createReadStream(path) : buffer;
```